src/plugins/intel_npu/src/plugin/npuw/llm_infer_request.cpp

```diff
@@ -0,0 +1,239 @@
+template <typename T>
+void fill_tensor(ov::Tensor tensor, T fill_val, size_t offset = 0u) {
+    T* tensor_data = tensor.data<T>();
+    std::fill(tensor_data + offset, tensor_data + tensor.get_size(), fill_val);
+}
+
+void copy_rows(const uint8_t* src,
+               const size_t num_rows,
+               const size_t row_byte_size,
+               const size_t stride,
+               uint8_t* dst) {
+    const uint8_t* src_row_p = src;
+    uint8_t* dst_row_p = dst;
+    for (int row = 0; row < num_rows; ++row) {
+        std::copy_n(src_row_p, row_byte_size, dst_row_p);
+        src_row_p += stride;
+        dst_row_p += stride;
+    }
+}
+
+void copy_to(ov::SoPtr<ov::ITensor> src,
+             ov::SoPtr<ov::ITensor> dst,
+             const size_t dim,
+             const size_t src_start,
+             const size_t dst_start,
+             const size_t count) {
+    // FIXME: Only 2 is supported now...
+    OPENVINO_ASSERT(dim == 2);
+    const auto* src_p = static_cast<uint8_t*>(src->data());
+          auto* dst_p = static_cast<uint8_t*>(dst->data());
+
+    const auto C  = src->get_shape()[1];
+    const auto SC = src->get_strides()[1];
+    const auto SH = src->get_strides()[dim];
+
+    for (int c = 0; c < C; ++c) {
+        const auto* sp = src_p + (C * SC) + (src_start * SH);
+              auto* dp = dst_p + (C * SC) + (dst_start * SH);
+        copy_rows(sp, count, SH, SH, dp);
+    }
+}
+
+ov::Tensor make_tensor_slice(ov::Tensor tensor, size_t dim, size_t start_pos, size_t end_pos) {
+    ov::Shape start_shape(std::vector<size_t>(tensor.get_shape().size(), 0u));
+    start_shape[dim] = start_pos;
+    ov::Shape end_shape = tensor.get_shape();
+    end_shape[dim] = end_pos;
+    return ov::Tensor(tensor, start_shape, end_shape);
+}
+
+ov::npuw::LLMInferRequest::LLMInferRequest(const std::shared_ptr<ov::npuw::LLMCompiledModel>& compiled_model,
+                                           const ov::npuw::LLMCompiledModel::KVCacheDesc& kvcache_desc)
+    : ov::ISyncInferRequest(compiled_model),
+      m_kvcache_desc(kvcache_desc) {
+    LOG_DEBUG("ov::npuw::LLMInferRequest::LLMInferRequest()");
+    LOG_BLOCK();
+    m_kvcache_request = compiled_model->kvcache_compiled->create_infer_request();
+    m_prefill_request = compiled_model->prefill_compiled->create_infer_request();
+
+    for (auto input_port : m_prefill_request->get_compiled_model()->inputs()) {
+        m_prefill_in_ports.emplace(input_port.get_any_name(), input_port);
+    }
+    for (auto output_port : m_prefill_request->get_compiled_model()->outputs()) {
+        m_prefill_out_ports.emplace(output_port.get_any_name(), output_port);
+    }
+
+    for (auto input_port : m_kvcache_request->get_compiled_model()->inputs()) {
+        m_kvcache_in_ports.emplace(input_port.get_any_name(), input_port);
+    }
+    for (auto output_port : m_kvcache_request->get_compiled_model()->outputs()) {
+        m_kvcache_out_ports.emplace(output_port.get_any_name(), output_port);
+    }
```

Comment:
Not sure what's the idea behind this, @TolyaTalamanov .. Normally we don't even need port names.